source/v5/mqtt3_to_mqtt5_adapter.c

```diff
@@ -1907,10 +1896,39 @@ struct aws_mqtt3_to_mqtt5_adapter_operation_subscribe *aws_mqtt3_to_mqtt5_adapte
     };
 
     subscribe_op->subscribe_op = aws_mqtt5_operation_subscribe_new(
-        allocator, options->adapter->client, &subscribe_view, &subscribe_completion_options);
+        allocator, subscribe_op->base.adapter->client, &subscribe_view, &subscribe_completion_options);
 
     if (subscribe_op->subscribe_op == NULL) {
-        goto error;
+        /* subscribe options validation will have been raised as the error */
+        return AWS_OP_ERR;
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+struct aws_mqtt3_to_mqtt5_adapter_operation_subscribe *aws_mqtt3_to_mqtt5_adapter_operation_new_subscribe(
+    struct aws_allocator *allocator,
+    const struct aws_mqtt3_to_mqtt5_adapter_subscribe_options *options) {
+
+    struct aws_mqtt3_to_mqtt5_adapter_operation_subscribe *subscribe_op =
+        aws_mem_calloc(allocator, 1, sizeof(struct aws_mqtt3_to_mqtt5_adapter_operation_subscribe));
+
+    subscribe_op->base.allocator = allocator;
+    aws_ref_count_init(&subscribe_op->base.ref_count, subscribe_op, s_adapter_subscribe_operation_destroy);
+    subscribe_op->base.impl = subscribe_op;
+    subscribe_op->base.type = AWS_MQTT3TO5_AOT_SUBSCRIBE;
+    subscribe_op->base.adapter = options->adapter;
+    subscribe_op->base.holding_adapter_ref = false;
+
+    /*
+     * If we're a regular subscribe, build the mqtt5 operation now.  Otherwise, we have to wait until
+     * we're on the event loop thread and it's safe to query the subscription set.
+     */
+    if (options->subscription_count > 0) {
```

Comment:
If it's possible to call `aws_mqtt_client_connection_subscribe_multiple` with an empty aws_array_list, the subscription_count would be zero, the subscribe_op->subscribe_op would remain null, and the subscribe multiple might trigger a resubscribe.

Reply:
True, will add a synchronous failure check for that case.